### ISSUE TITLE
Namespace 'NodeJS' has no exported member 'TypedArray'

### ISSUE THREAD

**Issue by JavierPAYTEF** [exceljs/exceljs#1015]
When compiling with typescript in a client enviroment it throws the following error.
I don't have any of the NodeJS types imported in the project because I don't need them, but it seems a couple are imported because of webpack.

Is this a bug with the definition?

To be had I do have the @types/node (version 12.12.5) folder because of a dependency, and I can't add that to the tsconfig because it generates all sort of problems with types in other parts of the application.

Removing NodeJS.TypedArray from the allowed types of dictionary fixes the issue.

> ERROR in D:/_dev/PAYTEFClientZone/client-zone/node_modules/exceljs/index.d.ts(1661,41):
> 
> 1661:41 Namespace 'NodeJS' has no exported member 'TypedArray'.
>     1659 |             memLevel: number; // compression only
>     1660 |             strategy: number; // compression only
> [x] 1661 |             dictionary: Buffer | NodeJS.TypedArray | DataView | ArrayBuffer; // deflate/inflate only, empty dictionary by default
>          |                                         ^
>     1662 |         }
>     1663 | 
>     1664 |              interface WorkbookStreamWriterOptions extends WorkbookWriterOptions {

**Comment by ahimik** [comment]
Got the same issue but resolved it by updating my devDependencies in `
package.json`
from `"@types/node": "^8.9.5"` to `"@types/node": "^12.12.7"` which is the lates one at the moment.

Verified with version 12.12.5 - works fine as well.

Make sure you are using the correct version of node types and verify that you have node types configured in your `tsconfig.json` file:
`"types": [
            "node",
            ...
]`

**Comment by JavierPAYTEF** [comment]
The issue is that not everybody needs or wants to include node types. In my case the application is for the browser, and nodejs adds definitions and types that I don't want, for example a definition for "setTimeout".

**Comment by Arios509** [comment]
@JavierPAYTEF But if in your case, when i import this function into the function, then will have  the error. According to your explanation, you mean added services could help to solve this problem?

**Comment by JavierPAYTEF** [comment]
@Arios509 thanks for the reply, what I meant is that importing the types for NodeJS globally make exceljs work but bring a whole bunch of issues elsewhere. And as I cited in my example, function "setTimeout" doesn't return a number anymore, because in NodeJS it returns a NodeJS.Timeout object, which breaks all places where I stored that value. This is one example of many.

Since I'm using this for a client application, and not backend, I don't necessarily want a nodejs types dependency.

Maybe adding a custom interface for an ExcelJS TypedArray would be better than depending on NodeJS version.

To be had, it appears this is the only NodeJS type that brings conflicts, I see a ton of other references to other NodeJS that don't bring issues. Maybe this type is part of a different definition, I'm not that knowledgeable about NodeJS inner types.
